std/algorithm/comparison.d

```diff
@@ -1911,3 +1915,97 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
         [mytuple(2, 3), mytuple(1, 2)]
     ));
 }
+
+enum implicitlyConvertsToBool(T) = isImplicitlyConvertible!(bool, T) || isArray!T;
```

Comment:
So this should be `ifTestable` and the implementation should just make sure that `if (T()) {}` compiles.